java/DEVELOPER.md

```diff
@@ -0,0 +1,260 @@
+-   openssl-dev
+-   rustup
+-   Java 11
+
+**Install protobuf compiler (necessary for all systems except for MacOS)**
+
+To install protobuf for MacOS, run:
+```bash
+brew install protobuf
+# Check that the protobuf compiler version 26.1 or higher is installed
+protoc --version
+```
+
+For the remaining systems, do the following:
+```bash
+PB_REL="https://github.com/protocolbuffers/protobuf/releases"
+curl -LO $PB_REL/download/v26.1/protoc-26.1-linux-x86_64.zip
+unzip protoc-26.1-linux-x86_64.zip -d $HOME/.local
+export PATH="$PATH:$HOME/.local/bin"
+# Check that the protobuf compiler version 26.1 or higher is installed
+protoc --version
+```
+
+**Dependencies installation for Ubuntu**
+
+```bash
+sudo apt update -y
+sudo apt install -y openjdk-11-jdk git gcc pkg-config openssl libssl-dev unzip
+# Install rust
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
+source "$HOME/.cargo/env"
+# Check that the Rust compiler is installed
+rustc --version
+```
+
+**Dependencies installation for CentOS**
+
+```bash
+sudo yum update -y
+sudo yum install -y java-11-openjdk-devel git gcc pkgconfig openssl openssl-devel unzip
+# Install rust
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
+```
+**Dependencies installation for MacOS**
+
+```bash
+brew update
+brew install openjdk@11 git gcc pkgconfig protobuf openssl protobuf
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
+source "$HOME/.cargo/env"
+```
+
+#### Building and installation steps
+
+Before starting this step, make sure you've installed all software dependencies.
+
+1. Clone the repository:
+    ```bash
+    VERSION=0.1.0 # You can modify this to other released version or set it to "main" to get the unstable branch
+    git clone --branch ${VERSION} https://github.com/aws/glide-for-redis.git
+    cd glide-for-redis/java
+    ```
+2. Initialize git submodule:
+    ```bash
+    git submodule update --init --recursive
+    ```
+3. Build the Java wrapper (Choose a build option from the following and run it from the `java` folder):
+
+    1. Enter the java directory:
+
+    ```bash
+    cd java
+    ```
+
+    2. Build in debug mode:
+
+    ```bash
+    ./gradlew :client:buildAll
+    ```
+
+    3. Build in release mode:
+
+    ```bash
+    ./gradlew :client:buildAllRelease
+    ```
+4. Install the rust linter
+    Rust
+    ```bash
+    # Run from the `java` folder
+    # will only need to run once during the installation process
+    rustup component add clippy rustfmt
+    cargo clippy --all-features --all-targets -- -D warnings
+    cargo fmt --manifest-path ./Cargo.toml --all
+    ```
+
+### Linters
+
+Development on the Java wrapper may involve changes in either the Java or Rust code. Each language has distinct linter tests that must be passed before committing changes.
+
+#### Language-specific Linters
+
+**Java:**
+
+-   Spotless
+
+#### Running the linters
+
+    Spotless
+    ```bash
+    # Run from the `java` folder
+    ./gradlew :spotlessApply
```

Comment:
`./gradlew :spotlessCheck` 
should be run first to see if any lintting changes are required